edk2basetools/VfrCompiler/IfrUtility.py

```diff
@@ -0,0 +1,2238 @@
+                    block_size += sizeof(EFI_HII_SIBT_DUPLICATE_BLOCK)
+                    current_string_id += 1
+
+            elif block_type == EFI_HII_SIBT_SKIP1:
+                skip_count = int.from_bytes(block_hdr[sizeof(EFI_HII_STRING_BLOCK)], byteorder="little")
+                current_string_id += skip_count
+                block_size += sizeof(EFI_HII_SIBT_SKIP1_BLOCK)
+
+            elif block_type == EFI_HII_SIBT_SKIP2:
+                skip_count = int.from_bytes(
+                    block_hdr[sizeof(EFI_HII_STRING_BLOCK) : sizeof(EFI_HII_STRING_BLOCK) + sizeof(ctypes.c_uint16)],
+                    byteorder="little",
+                )
+                current_string_id += skip_count
+                block_size += sizeof(EFI_HII_SIBT_SKIP2_BLOCK)
+
+            elif block_type == EFI_HII_SIBT_EXT1:
+                length8 = int.from_bytes(
+                    block_hdr[sizeof(EFI_HII_STRING_BLOCK) + sizeof(ctypes.c_uint8) : sizeof(EFI_HII_STRING_BLOCK) + sizeof(ctypes.c_uint8) + sizeof(ctypes.c_uint8)],
+                    byteorder="little",
+                )
+                block_size += length8
+
+            elif block_type == EFI_HII_SIBT_EXT2:
+                ext2 = EFI_HII_SIBT_EXT2
+                length32 = int.from_bytes(
+                    block_hdr[sizeof(EFI_HII_STRING_BLOCK) + sizeof(ctypes.c_uint8) : sizeof(EFI_HII_STRING_BLOCK) + sizeof(ctypes.c_uint8) + sizeof(ctypes.c_uint32)],
+                    byteorder="little",
+                )
+                block_size += length32
+
+            if string_id > 0 and string_id != (ctypes.c_uint16)(-1):
+                string_text_offset = block_hdr - string_data + offset
+                block_type = block_hdr[0]
+
+                if string_id == current_string_id - 1:
+                    if block_type == EFI_HII_SIBT_SKIP2 or block_type == EFI_HII_SIBT_SKIP1:
+                        return EFI_NOT_FOUND
+                    else:
+                        return EFI_SUCCESS
+
+                if string_id < current_string_id - 1:
+                    return EFI_NOT_FOUND
+
+            block_hdr = string_data + block_size
+
+        return EFI_NOT_FOUND
+
+    def GetVarStoreNameFormStringId(self, StringId):
+        pInFile = None
+        NameOffset = 0
+        Length = 0
+        StringPtr = None
+        StringName = None
+        UnicodeString = None
+        VarStoreName = None
+        DestTmp = None
+        Current = None
+        Status = None
+        LineBuf = [0] * EFI_IFR_MAX_LENGTH
+        BlockType = 0
+        PkgHeader = None
+
+        if self.StringFileName is None:
+            return None
+
+        try:
+            pInFile = open(LongFilePath(self.StringFileName), "rb")
+            # Get file length
+            pInFile.seek(0, os.SEEK_END)
+            Length = pInFile.tell()
+            pInFile.seek(0, os.SEEK_SET)
+
+            # Get file data
+            StringPtr = bytearray(pInFile.read())
+        except:
```

Comment:
do not use a bare except.
```suggestion
        except Exception:
```